app/javascript/packs/dashboardTags.js

```diff
@@ -0,0 +1,178 @@
+import { initializeDropdown } from '@utilities/dropdownUtils';
+
+listenForButtonClicks();
+
+/**
+ * Adds an event listener to the inner page content, to handle any and all follow button clicks with a single handler
+ */
+function listenForButtonClicks() {
+  document
+    .getElementById('following-wrapper')
+    .addEventListener('click', handleClick);
+}
+
+/**
+ * Checks a click event's target to see which button was clicked and calls the relevant handlers
+ *
+ * @param {HTMLElement} target The target of the click event
+ */
+function handleClick({ target }) {
+  const tagContainer = target.closest('.dashboard__tag__container');
+
+  if (target.classList.contains('follow-button')) {
+    handleFollowingButtonClick(tagContainer);
+  }
+
+  if (target.classList.contains('hide-button')) {
+    handleHideButtonClick(tagContainer);
+  }
+
+  if (target.classList.contains('unhide-button')) {
+    handleUnhideButtonClick(tagContainer);
+  }
+}
+
+function fetchFollows(body) {
+  const tokenMeta = document.querySelector("meta[name='csrf-token']");
+  const csrfToken = tokenMeta && tokenMeta.getAttribute('content');
+
+  return window.fetch('/follows', {
+    method: 'POST',
+    headers: {
+      'X-CSRF-Token': csrfToken,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(body),
+    credentials: 'same-origin',
+  });
+}
+
+function handleFollowingButtonClick(tagContainer) {
+  const { tagId, followId } = tagContainer.dataset;
+
+  const data = {
+    followable_type: 'Tag',
+    followable_id: tagId,
+    verb: 'unfollow',
+  };
+
+  fetchFollows(data)
+    .then(() => {
+      removeElementFromPage(followId);
+      updateNavigationItemCount();
+    })
+    .catch((error) => console.error(error));
+}
+
+function handleHideButtonClick(tagContainer) {
+  const { tagId, followId } = tagContainer.dataset;
+
+  const data = {
+    followable_type: 'Tag',
+    followable_id: tagId,
+    verb: 'follow',
+    explicit_points: -1,
+  };
+
+  fetchFollows(data)
+    .then(() => {
```

Comment:
In the success (maybe even in the error case) case we want to add a Snackbar.  Snackbars are done in React and our this file is vanilla js which makes it challenging to integrate. I. noticed we sometimes used custom events `snackbar:add` but its been taking some time to figure this out, hence I'll implement the snackbar in a subsequent PR so that we can still release iteratively.